### ISSUE TITLE
Why do the masks' positions on the visualized images are same?

### ISSUE THREAD

**Issue by WhiteLie98** [IcarusWizard/MAE#5]
I run the pretrain program and saw the visualized images on tensorboard. I found that the masks of those 16 images are exactly the same as yours. So I was wondering if it isn't actually random shuffled?
The picture with time is the visualization of mine, and the other is yours.
![问题1](https://user-images.githubusercontent.com/39949809/146378844-1c3e99ab-bcab-440f-a861-27198cbfd5c8.png)
![问题2](https://user-images.githubusercontent.com/39949809/146378864-23e6169d-f09d-4c88-8533-f9b12bfba78e.png)

**Comment by IcarusWizard** [comment]
Yes, these images are just the first 16 images on the cifar10 validation dataset. I didn't shuffle the dataset since I want to see how reconstruction gets better as we trained longer.

**Comment by WhiteLie98** [comment]
抱歉，可能我表达的不是很清楚，我的疑惑是为什么mask的位置都一样？按理论讲应该是随机遮掩的吧，我看model的patchshuffle代码部分好像也没问题，所以不太懂😂



---Original---
From: ***@***.***&gt;
Date: Thu, Dec 16, 2021 22:18 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [IcarusWizard/MAE] Why do the masks on visualized dataset aresame? (Issue #5)




 
Yes, these images are just the first 16 images on the cifar10 validation dataset. I didn't shuffle the dataset since I want to see how reconstruction gets better as we trained longer.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by WhiteLie98** [comment]
> Yes, these images are just the first 16 images on the cifar10 validation dataset. I didn't shuffle the dataset since I want to see how reconstruction gets better as we trained longer.

Sorry, maybe I didn't express the question clearly. My question is not about the selected 16 images from the dataset, but the reason why are those masks positions same? Theoretically, these masked patches should be shuffled randomly.

**Comment by IcarusWizard** [comment]
Oh, I see. 

Yes, the patches are shuffled randomly, and you can verify that by checking the visualization for different epochs (just scroll the tensorboard).

However, you get the same result as mine is because we fix the random seed at the beginning, which controls all the randomness. That is, if you run the code without changing the default random seed, you get exactly the same result I got, even for the positions that were masked in the last visualization.

**Comment by WhiteLie98** [comment]
> Oh, I see.
> 
> Yes, the patches are shuffled randomly, and you can verify that by checking the visualization for different epochs (just scroll the tensorboard).
> 
> However, you get the same result as mine is because we fix the random seed at the beginning, which controls all the randomness. That is, if you run the code without changing the default random seed, you get exactly the same result I got, even for the positions that were masked in the last visualization.

I see. Thank you very much!
